wordpress-mobile/WordPress-Android
# Stats revamp v2 views and visitors graph improvements

This PR fixes all the outstanding issues on Views and Visitors graph from the following PRs
- [Views and Visitors Card](https://github.com/wordpress-mobile/WordPress-Android/pull/16258)
- [Views and Visitors Graph](https://github.com/wordpress-mobile/WordPress-Android/pull/16327)

Issues Fixed:
- Tool tip (Marker) now morphs to top, bottom, left and right based on location of the data point within graph
- Tool tip only appears only for current week line data
- data point circle only appears along with marker
- Shows ticks for data points on xAxis
- dates on xAxis are formatted better
- Copy of information line is updated from "Your this week's views ..". to "Your views this week"
- Chips are updated to correct colour
- Refactor and cleaned up

Fixes #16274 

| Marker | Marker | Marker
|---|---|---|
| ![Screenshot_20220517_001839](https://user-images.githubusercontent.com/990349/168614649-0b60002c-febc-44ab-826e-c24dd35cde19.png)|![Screenshot_20220517_001734](https://user-images.githubusercontent.com/990349/168614770-25a98792-e64f-4e1e-91ac-41317ddd7ad7.png)| ![Screenshot_20220517_001753](https://user-images.githubusercontent.com/990349/168614846-21afe92d-3d2e-47ef-8caf-e2d3ce300760.png)|


To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab as in the images above (scroll down if necessary)

Test 2:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab with Line Chart as in the images above (scroll down if necessary)
- Notice that the Views graph is in blue and this week and previous week and percent change are all in order
- Notice the graph, data and change are all compare this week with previous week
- Switch to Visitors by tapping chip and notice graph is in purple color

Test 3:
- Launch app
- Go to stats either using quick links or menu 
- Notice a new **Views & Visitors** card is shown in Insights tab with Line Chart as in the images above (scroll down if necessary)
- Tap on the graph.  Notice a marker popup as in the third image above.


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Existing tests

PR submission checklist:
